lib/controllers/share.js

```diff
@@ -52,16 +69,32 @@ function resolveGist(serviceOptions, id) {
     if (serviceOptions.accessToken !== undefined) {
         headers['Authorization'] = 'token ' + serviceOptions.accessToken;
     }
-    return requestp({
+    return got({
         url: gistAPI + '/' + id,
         headers: headers,
-        json: true,
-        transform: function(body, response) {
-            if (response.statusCode >= 300) {
-                return response;
-            } else {
-                return body.files[Object.keys(body.files)[0]].content; // find the contents of the first file in the gist
-            }
+        responseType: 'json',
+        hooks: {
+            afterResponse: [
+                (response, _retryWithMergedOptions) => {
+                    if (response.statusCode >= 300) {
+                        return response;
+                    } else {
+                        return response.body.files[Object.keys(response.body.files)[0]].content; // find the contents of the first file in the gist
```

Comment:
Got seems pretty compatible with request, many options have the same names.

However, I don't think this is tested by the test suite, and I have no idea if it is correct with got.